camel/configs/gemini_config.py

```diff
@@ -14,22 +14,19 @@
 
 
 from collections.abc import Iterable
-from dataclasses import asdict, dataclass
-from typing import TYPE_CHECKING, Optional
+from typing import Optional
 
-from camel.configs.base_config import BaseConfig
+from google.generativeai.protos import Schema
+from google.generativeai.types.content_types import (
+    FunctionLibraryType,
+    ToolConfigType,
+)
+from google.generativeai.types.helper_types import RequestOptionsType
+from google.generativeai.types.safety_types import SafetySettingOptions
 
-if TYPE_CHECKING:
-    from google.generativeai.protos import Schema
-    from google.generativeai.types.content_types import (
-        FunctionLibraryType,
-        ToolConfigType,
-    )
-    from google.generativeai.types.helper_types import RequestOptionsType
-    from google.generativeai.types.safety_types import SafetySettingOptions
```

Comment:
we need to remove TYPE_CHECKING to specify the attributes type in pydantic.
The `TYPE_CHECKING ` default to false, it only set to True when mypy checking.
If we remove it, the pydantic will raise below errror, e.g :
```
E       pydantic.errors.PydanticUserError: `GroqConfig` is not fully defined; you should define `OpenAIFunction`, then call `GroqConfig.model_rebuild()`.
E       
E       For further information visit https://errors.pydantic.dev/2.6/u/class-not-fully-defined
```

Reply:
The issue here is it's making optional packages to be required, if user just install basic camel using poetry install, they will not able to be run any basic example file like examples/single_agent.py and examples/ai_society/role_playing.py, any better way to fix this?

Reply:
I think the optional packages by TYPE_CHECKING is the problem. 
Why we want to use pydantic ? Because we want to define exactly what the data type is while not dynamical way.
We need to do some research to find the trade off